dqc_us_rules/dqc_us_0001.py

```diff
@@ -26,6 +28,92 @@
 _UGT_FACT_KEY = 'ugt_fact'
 _NO_FACT_KEY = 'no_fact'
 _EXT_FACT_KEY = 'ext_fact'
+_EARLIEST_US_GAAP_YEAR = 2014
+
+ugtDocs = (
+    {
+        "year": 2014,
+        "namespace": "http://fasb.org/us-gaap/2014-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/elts/us-gaap-doc-2014-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/entire/us-gaap-entryPoint-std-2014-01-31.xsd",  # noqa
+    },
+    {
+        "year": 2015,
+        "namespace": "http://fasb.org/us-gaap/2015-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/elts/us-gaap-doc-2015-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/entire/us-gaap-entryPoint-std-2015-01-31.xsd",  # noqa
+    },
+    {
+        "year": 2016,
+        "namespace": "http://fasb.org/us-gaap/2016-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/elts/us-gaap-doc-2016-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/entire/us-gaap-entryPoint-std-2016-01-31.xsd",  # noqa
+    },
+)
+
+
+def _create_config(val):
+    """
+    Creates the configs needed for dqc_us_0001
+
+    :param val: ValidateXbrl needed in order to save the cache
+    :type val: :class: '~arelle.ValidateXbrl.ValidateXbrl'
+    :return: no explicit return but creates and saves configs in
+        dqc_us_rule\resources\DQC_US_0001
+    :rtype: None
+    """
+    val.ugtNamespace = None
+    cntlr = val.modelXbrl.modelManager.cntlr
+    year = _EARLIEST_US_GAAP_YEAR
+    config = _load_config(_DQC_01_AXIS_FILE)
+    # Create a list of axes in the base config file
+    axisMembers = set()
+    # receives list of members of above axes
+
+    def traverseMembers(axisConceptName, parentModelObject,
+                        relName, ELR):
+        for rel in dimLoadingInstance.relationshipSet(relName, ELR).fromModelObject(parentModelObject):  # noqa
```

Comment:
This line needs to be broken up. Consider making some variables to shorten the line